[feature] support fluentd_pid_dir and fluentd_pid_file

trombik opened the pull request:
fixes #38

Commit:
support fluentd_pid_dir and fluentd_pid_file

fixes #38

Commit:
fix centos build

Commit:
fix ubuntu build

Commit:
do not run assert tasks during check mode

because, when check mode is enabled by `-C` and the target node has not
been converged by ansible, the assertions fail.

Commit:
QA

mitsururike commented on README.md:
```diff
@@ -66,6 +68,8 @@ Note that although the role creates `fluentd_log_dir`, you need to configure
 | `__fluentd_gem_bin` | `/usr/sbin/td-agent-gem` |
 | `__fluentd_unix_pipe_dir` | `/var/tmp/fluentd` |
 | `__fluentd_flags` | `""` |
+| `__fluentd_pid_dir` | `/var/run/fluentd` |
```
/fluentd -> /td-agent

mitsururike commented on README.md:
```diff
@@ -66,6 +68,8 @@ Note that although the role creates `fluentd_log_dir`, you need to configure
 | `__fluentd_unix_pipe_dir` | `/var/tmp/fluentd` |
 | `__fluentd_flags` | `""` |
+| `__fluentd_pid_dir` | `/var/run/fluentd` |
+| `__fluentd_pid_file` | `{{ fluentd_pid_dir }}/fluentd.pid` |
```
/fluentd -> /td-agent

mitsururike commented on README.md:
```diff
@@ -96,6 +102,9 @@ Note that although the role creates `fluentd_log_dir`, you need to configure
 | `__fluentd_unix_pipe_dir` | `/var/tmp/fluentd` |
 | `__fluentd_flags` | `""` |
+| `__fluentd_pid_dir` | `/var/run/td-agent` |
+| `__fluentd_pid_file` | `{{ fluentd_pid_dir }}/fluentd.pid` |
```
/fluentd -> /td-agent

mitsururike commented on tasks/main.yml:
```diff
@@ -143,3 +143,30 @@
     enabled: true
     state: started
+
+- name: Check if pid directory exists
```
file module with state 'directory' is simpler.

mitsururike commented on tasks/main.yml:
```diff
@@ -143,3 +143,30 @@
+      - register_fluentd_stat_pid_dir.stat.isdir
+  when: not ansible_check_mode
+
+- name: Check if pid file exists
```
some explanation should be written why fluentd fails to create pid file.
restarting fluentd does not solve the problem?

trombik replied:
as commented in cb74d94cd09b1a671f1b7780fd7f4036b423927d, they are a test in `ansible` play. `serverspec` helps role development a lot but does not when used in a project. we should have [more tests in roles](http://docs.ansible.com/ansible/test_strategies.html) as well.

Commit:
sync

Commit:
use `state: file` and explain the rationale

https://github.com/reallyenglish/ansible-role-fluentd/pull/39#discussion_r117668444
https://github.com/reallyenglish/ansible-role-fluentd/pull/39#discussion_r117668773
